src/x11/x.c

```diff
@@ -154,15 +135,34 @@ static void x_win_round_corners(struct window_x11 *win, const int rad)
 
 void x_display_surface(cairo_surface_t *srf, struct window_x11 *win, const struct dimensions *dim)
 {
+        char astr[sizeof("_NET_WM_CM_S") / sizeof(char) + 8];
+        Atom cm_sel;
+
         x_win_move(win, dim->x, dim->y, dim->w, dim->h);
         cairo_xlib_surface_set_size(win->root_surface, dim->w, dim->h);
 
+        XClearWindow(xctx.dpy, win->xwin);
+        XFlush(xctx.dpy);
+
         cairo_set_source_surface(win->c_ctx, srf, 0, 0);
         cairo_paint(win->c_ctx);
         cairo_show_page(win->c_ctx);
 
-        if (settings.corner_radius != 0)
+        sprintf(astr, "_NET_WM_CM_S%i", win->cur_screen);
+        cm_sel = XInternAtom(xctx.dpy, astr, true);
+        if (settings.corner_radius != 0 && XGetSelectionOwner(xctx.dpy, cm_sel) == None)
```

Comment:
This should probably be a new function `bool has_compositing()` or something similar, keep things simple